misc/server/src/main.rs

```diff
@@ -0,0 +1,213 @@
+use base64::Engine;
+use clap::Parser;
+use futures::executor::block_on;
+use futures::future::Either;
+use futures::stream::StreamExt;
+use futures_timer::Delay;
+use libp2p::core::muxing::StreamMuxerBox;
+use libp2p::core::upgrade;
+use libp2p::dns;
+use libp2p::identify;
+use libp2p::identity;
+use libp2p::identity::PeerId;
+use libp2p::kad;
+use libp2p::metrics::{Metrics, Recorder};
+use libp2p::noise;
+use libp2p::swarm::{SwarmBuilder, SwarmEvent};
+use libp2p::tcp;
+use libp2p::yamux;
+use libp2p::Transport;
+use log::{debug, info};
+use prometheus_client::metrics::info::Info;
+use prometheus_client::registry::Registry;
+use std::error::Error;
+use std::io;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::task::Poll;
+use std::thread;
+use std::time::Duration;
+use zeroize::Zeroizing;
+
+mod behaviour;
+mod config;
+mod http_service;
+
+const BOOTSTRAP_INTERVAL: Duration = Duration::from_secs(5 * 60);
+
+#[derive(Debug, Parser)]
+#[clap(name = "libp2p server", about = "A rust-libp2p server binary.")]
+struct Opts {
+    /// Path to IPFS config file.
+    #[clap(long)]
+    config: PathBuf,
+
+    /// Metric endpoint path.
+    #[clap(long, default_value = "/metrics")]
+    metrics_path: String,
```

Comment:
Why is it important that this is configurable?

Reply:
Depends on the deployment. The Kubo IPFS bootstrappers listen on `/debug/metrics/prometheus`. The Prometheus default is `/metrics`. I want to make it easy on our operations team without sacrificing the benefit of the default across the Prometheus ecosystem.